core/src/main/resources/defaults/blueprints/cdp-data-science.bp

```diff
@@ -152,6 +152,69 @@
             "base": true
           }
         ]
+      },
+      {
+        "refName": "hue",
+        "serviceType": "HUE",
+        "roleConfigGroups": [
+          {
+            "refName": "hue-HUE_SERVER-BASE",
+            "roleType": "HUE_SERVER",
+            "base": true,
+            "configs": [
+              {
+                "name": "log_directory_free_space_percentage_thresholds",
```

Comment:
Are these directory free space configs required?  If so, they should probably be generated by CB for each role for which they are needed.  Otherwise I think we should omit them.

Reply:
Let me interject on behalf of Denes: According to [the online Hue docs](https://www.cloudera.com/documentation/enterprise/6/properties/6.2/topics/cm_props_cdh620_hue.html), these alert settings default to 10GB/5GB absolute thresholds, so it probably makes sense to override them in terms of percentage.

Reply:
Thanks for the pointer.  The same is true for other services (eg. YARN, Spark, even ZooKeeper).  That's why I think it should be automatically added by CB instead of repeating the same stuff for each service/role in the template.

Reply:
Having these in Java and added automatically per service/role would indeed have a great advantage: Self-constructed templates (i.e. a la carte via UI or CLI) would automatically benefit from these extra settings when the respective services are present.

On the other hand, I wonder how config overriding would happen in that case: What if the user wanted to apply different settings for these properties? In particular, the current implementation of the respective classes (`CmTemplateComponentConfigProcessor`, `CmTemplateProcessor` and the SDK `ApiClusterTemplate`) do not seem to be able to handle this, and would happily allow a given property be specified multiple times in the final generated template. That will either result in a template processing error in CM, or the last occurrence of the property will win.

Reply:
I think we should change the behaviour and do not override values that are present in the template so users can customize their templates.

Reply:
so I did not get the final conclusion on this... shall I add these props to the template -or- wire it into the code.
BTW. based on Devaraj's request, the Hue (and Oozie) must be included in both the Data Engineering and into the Data Mart templates.

Reply:
My suggestion is to omit these from the template, and add them later in a separate task, preferably to the code.  They are neither specific to, nor essential for Hue.